File: JAVA_PROGRAMMING/20_10_2021/PrimeNumber/src/com/apisero/greenfield/Prime.java
package com.apisero.greenfield;

public class Prime 
{
	int num;
	Prime(int num)
	{
		this.num = num;
	}
	void checkPrime()
    {
        int i=2;
        if(this.num==1)
        {
            System.out.printf("%d is not a prime number.\n",this.num);
        }
        while(i<=this.num)
        {
            
            if(this.num%i==0)
            {
                if(this.num==i)
                {
                	System.out.printf("%d is a prime number.\n",this.num);   
                }
                else
                {
                	System.out.printf("%d is not a prime number.\n",this.num);
                    break;
                }
            }
            i++;
        }
    }
	public static void main(String[] args) 
	{
		Prime p = new Prime(10);
		p.checkPrime();
		Prime p1 = new Prime(11);
		p1.checkPrime();
		Prime p2 = new Prime(17);
		p2.checkPrime();
	}

}

File: CPP_PROGRAMMING/20_10_2021/prime.cpp
#include<iostream>
using namespace std;

class Prime
{
    int num;
public:
    Prime(int num)
    {
        this->num = num;
    }
    void checkPrime()
    {
        int i=2;
        if(this->num==1)
        {
            printf("%d is not a prime number.\n",this->num);
        }
        while(i<=this->num)
        {
            
            if(this->num%i==0)
            {
                if(this->num==i)
                {
                printf("%d is a prime number.\n",this->num);   
                }
                else
                {
                    printf("%d is not a prime number.\n",this->num);
                    break;
                }
            }
            i++;
        }
    }
};

int main()
{
    Prime p(2);
    Prime q(28);
    p.checkPrime();
    q.checkPrime();
    
    return 0;
}
